src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.html

```diff
@@ -1,160 +1,32 @@
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="maxTrainWeightTemplate"
-    [template]="maxTrainWeightTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="axleWeightsTemplate"
-    [template]="axleWeightsTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="tyresTemplate"
-    [template]="tyresTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="brakesTemplate"
-    [template]="brakesTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="purchasersTemplate"
-    [template]="purchasersTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="dimensionsTemplate"
-    [template]="dimensionsTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="firstMinMaxTemplate"
-    [template]="firstMinMaxTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="secondMinMaxTemplate"
-    [template]="secondMinMaxTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="platesTemplate"
-    [template]="platesTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="trlAuthIntoServiceTemplate"
-    [template]="trlAuthIntoServiceTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
-  <app-dynamic-form-group
-    *ngIf="trlManufacturerTemplate"
-    [template]="trlManufacturerTemplate"
-    [data]="vehicleTechRecordCalculated"
-    class="govuk-summary-list"
-  ></app-dynamic-form-group>
+  <app-accordion-control>
+    <ng-container *ngFor="let section of sectionTemplates" [ngSwitch]="section.name">
+      <app-accordion *ngIf="section.name !== 'reasonForCreationSection' || isEditable" [id]="section.name" [title]="section.label">
+        <app-dimensions
+          *ngSwitchCase="'dimensionsSection'"
+          [vehicleTechRecord]="vehicleTechRecordCalculated"
+          [isEditing]="isEditable"
+          (formChange)="handleFormState($event)"
+        ></app-dimensions>
+
+        <app-dynamic-form-group
+          *ngSwitchCase="'brakes'"
+          [edit]="isEditable"
+          [template]="section"
+          [data]="vehicleTechRecordCalculated.brakes"
```

Comment:
Isn't is possible to change the brakes template to not require on the brakes data to be passed in? E.g.: mirror the structure of tech record component in the template?

Reply:
I think so, I think this was done early and never refactored

Reply:
Maybe leave it here tho @tahayazici and do it when you do the break section ammending haha